### DISCUSSION TITLE
Contradictory messages from brew link dry-run

### DISCUSSION THREAD

**Discussion by givihuda** [Homebrew/discussions#1317] [Everyday usage]
So, I wanted to see which files `brew link` would update for different PHP versions (the same ones I think, but I wanted to see the list since learning of the --dry-run option), but when I try to see, I get contradictory messages.

I know there were 25 files that were updated, so when I ran the dry run, it lists them until one has an error... and then it appears to halt. The problem, now I can't see what all it would update without apparently unlinking... which defeats the purpose of a dry run for a version that isn't linked at the moment (why unlink if I don't need to?).

```
MacBook-Pro ~ % brew link --dry-run php@7.3 
Would link:
/usr/local/bin/pear
/usr/local/bin/peardev
/usr/local/bin/pecl
/usr/local/bin/phar
/usr/local/bin/phar.phar
/usr/local/bin/php
/usr/local/bin/php-cgi
/usr/local/bin/php-config
/usr/local/bin/phpdbg
/usr/local/bin/phpize
/usr/local/sbin/php-fpm
Error: Could not symlink include/php/TSRM/TSRM.h
Target /usr/local/include/php/TSRM/TSRM.h
is a symlink belonging to php@5.6. You can unlink it:
  brew unlink php@5.6

To force the link and overwrite all conflicting files:
  brew link --overwrite php@5.6

To list all files that would be deleted:
  brew link --overwrite --dry-run php@5.6
```

Note the suggestion... to try a dry run of overwrite link of the existing linked version (php@5.6)... but a dry run on this tells me that it's already linked and doesn't tell me which files would be changed:

```
My-MacBook-Pro ~ % brew link --overwrite --dry-run php@5.6
Warning: Already linked: /usr/local/Cellar/php@5.6/5.6.40
To relink:
  brew unlink php@5.6 && brew link --force php@5.6
```

So, now I'm a circle, and unable to see what all running a dry link of php@7.3 would do, or even what a dry run overwrite of the existing version would change (so, two issues now, really), without having to first unlink php@5.6... I don't see the logical point of having to unlink just to see these dry-runs.

I hope what I'm saying makes sense... I think it's feasible to want to see which files would be updated on a dry run of a certain version without having to unlink the existing version.

**Comment by SMillerDev** [comment]
`brew link` doesn't automatically unlink other software so you would always have to unlink before linking. In your case the dry run is finding conflicts with php 5.6 and trying to give some advice on how to find files that would conflict.

Since `link` and `unlink` are atomic you can just unlink one formula and dry-run link the other one. It's actually the system that `brew-php-switcher` is based on.
